e2etest/l2tests/interface_selector.go

```diff
@@ -173,5 +174,58 @@ var _ = ginkgo.Describe("L2-interface selector", func() {
 				}, 10*time.Second, 1*time.Second).Should(gomega.Not(gomega.HaveOccurred()))
 			}
 		})
+
+		ginkgo.It("Specify not existing interfaces", func() {
+			resources := internalconfig.ClusterResources{
+				L2Advs: []metallbv1beta1.L2Advertisement{
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: "with-interfaces",
+						},
+						Spec: metallbv1beta1.L2AdvertisementSpec{
+							Interfaces: []string{"foo"},
+						},
+					},
+				},
+			}
+			err := ConfigUpdater.Update(resources)
+			framework.ExpectNoError(err)
+
+			svc, _ := service.CreateWithBackend(cs, f.Namespace.Name, "lb-service")
+			defer func() {
+				err := cs.CoreV1().Services(svc.Namespace).Delete(context.TODO(), svc.Name, metav1.DeleteOptions{})
+				framework.ExpectNoError(err)
+			}()
+
+			ingressIP := e2eservice.GetIngressPoint(&svc.Status.LoadBalancer.Ingress[0])
+			err = mac.FlushIPNeigh(ingressIP, executor.Host)
+			framework.ExpectNoError(err)
+
+			// check arp respond
+			for i := range LocalNics {
+				gomega.Consistently(func() error {
+					return mac.RequestAddressResolutionFromIface(ingressIP, LocalNics[i], executor.Host)
+				}, 10*time.Second, 1*time.Second).Should(gomega.HaveOccurred())
+			}
+
+			// check announceFailed event
+			gomega.Eventually(func() error {
+				events, err := cs.CoreV1().Events(svc.Namespace).List(context.Background(), metav1.ListOptions{FieldSelector: "reason=announceFailed"})
+				if err != nil {
+					return err
+				}
+
+				var svcMsg string
+				for _, e := range events.Items {
+					if e.InvolvedObject.Name == svc.Name {
+						svcMsg = e.Message
+					}
+				}
+				if svcMsg == "" {
```

Comment:
and return the error if you exit the loop

Reply:
done